### PR TITLE
mangademon extension

### PR DESCRIPTION
**Pull request by GandalfTheGoat**
Closes: #16078

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        }
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = mutableListOf<Page>()
+        document.select("img.imgholder").forEachIndexed { i, it ->
+            pages.add(Page(i, "", it.attr("src")))
+        }
+        return pages
+    }
```
Comment (alessandrojean) [inline src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt:132]:
```suggestion
    override fun pageListParse(document: Document): List<Page> {
        return document.select("img.imgholder")
            .mapIndexed { i, el -> Page(i, "", el.attr("src") }
    }
```

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MangaDemon : ParsedHttpSource() {
+
+    override val lang = "en"
+    override val supportsLatest = true
+    override val name = "MangaDemon"
```
Comment (alessandrojean) [inline src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt:21]: The site names itself "Manga Demon", so maybe follow it?

```suggestion
    override val name = "Manga Demon"
```

File: src/en/mangademon/build.gradle [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+ext {
+    extName = 'MangaDemon'
```
Comment (alessandrojean) [inline src/en/mangademon/build.gradle:5]:
```suggestion
    extName = 'Manga Demon'
```

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    override val baseUrl = "https://mangademon.org"
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("referrer", "origin")
+        .add("googlebot", "noindex")
+        .add("robots", "noindex")
```
Comment (alessandrojean) [inline src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt:27]: Is this really needed?

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    override fun searchMangaNextPageSelector() = latestUpdatesNextPageSelector()
+
+    // Manga details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val infoElement = document.select("article")
+
+        return SManga.create().apply {
+            title = infoElement.select("h1.novel-title").text()
+            author = infoElement.select("div.author").text().drop(7)
+            artist = ""
```
Comment (alessandrojean) [inline src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt:86]: Can just not set if the site doesn't provide it.

```suggestion
```

**Review by alessandrojean — CHANGES_REQUESTED** [review]

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+    override fun popularMangaNextPageSelector() = latestUpdatesNextPageSelector()
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    override fun popularMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+
+    // Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return POST("$baseUrl/search.php?manga=$query", headers)
```
[the remark this answers is not in the document]
Comment (GandalfTheGoat) [reply src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt]: I have implemented this.

File: src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListSelector() = "ul.chapter-list li"
+
+    // Get Chapters
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            element.select("a").let { urlElement ->
+                url = (urlElement.attr("href"))
```
[the remark this answers is not in the document]
Comment (GandalfTheGoat) [reply src/en/mangademon/src/eu/kanade/tachiyomi/extension/en/mangademon/MangaDemon.kt:108]: it uses relative URLs not absolute URLs for all of the links

**Pull request merged**